checker/src/call_visitor.rs

```diff
@@ -1035,31 +1035,29 @@ impl<'call, 'block, 'analysis, 'compilation, 'tcx, E>
     #[logfn_inputs(TRACE)]
     fn handle_add_tag(&mut self) {
         precondition!(self.actual_args.len() == 1);
-        if let Some(tag) = self.extract_tag_kind_and_mask() {
+
+        if let Some(tag) = self.extract_tag_kind_and_propagation_set() {
```

Comment:
You can deconstruct tag here so that you don't have to do tag.0 and tag.1 below.

Reply:
`tag.0` and `tag.1` are only used in logging, and `tag` as a whole is used later in the code.

Reply:
The logging is likely remain in the code, so it may as well be as readable as possible.

Reply:
Changed the type of tags from pairs to a structs. Now the fields are named.